src/kernels/MIOpenBatchNormFwdTrainSpatial.cl

```diff
@@ -329,13 +350,23 @@ MIOpenBatchNormFwdTrainSpatial(const __global _FLOAT* __restrict in,
     pvscale = lcl_scale;
     pvbias  = lcl_bias;
 
-#if(MIO_BN_REM == 0)
-    __attribute__((opencl_unroll_hint(2))) for(unsigned int k = lid; k < MIO_BN_LESS;
+#if(MIO_LAYOUT_NHWC == 1 || MIO_BN_REM == 0)
+    __attribute__((opencl_unroll_hint(2))) for(unsigned int k = lid; k <
+#if MIO_LAYOUT_NHWC == 1
+                                                                     MIO_BN_NHW
+#else
+                                                                     MIO_BN_LESS
+#endif
+                                               ;
                                                k += MIO_BN_GRP0)
```

Comment:
```suggestion
    const unsigned int k_limit =
#if MIO_LAYOUT_NHWC == 1
                                                                     MIO_BN_NHW;
#else
                                                                     MIO_BN_LESS;
#endif
    __attribute__((opencl_unroll_hint(2))) for(unsigned int k = lid; k < k_limit; k += MIO_BN_GRP0)
```

Reply:
@DrizztDoUrden is this change critical? How about we maintain it as-is?

Reply:
recommendations adopted